### PR TITLE
Remove deprecated `crypto` APIs

### LINKED ISSUE #38: Replace deprecated Crypto API calls
New OTP version deprecated the `crypto:private_encrypt()` usage. We need to upgrade the library.

### PR DESCRIPTION
**Pull request by whatyouhide**
This PR is based off of #36 and closes #38.

The issue was that `crypto:sign/5` **hashes** the input before signing it. It's enough to pass in `none` as the digest algorithm, and we get back the behavior of `crypto:public_encrypt/4`. The same goes for decrypting (with `crypto:verify` now).

### CODE REVIEW

**Comment by weppos** [comment]
Switching to @DXTimer , @m0rcq , and/or @san983 as reviewers as they focused on this task, and also know what else to cleanup upon completion of this blocker.

**Review by DXTimer — APPROVED** [review]
Nicely done. Everything works as expected and the API stays compatible with erldns.

**Comment by whatyouhide** [comment]
@DXTimer thanks for the review! 💟 

@weppos anything that needs to be done in this PR, or I can go ahead and merge it myself?

**Review by m0rcq — APPROVED** [review]

**Comment by weppos** [comment]
> @weppos anything that needs to be done in this PR, or I can go ahead and merge it myself?

It seems good to go.

@DXTimer do we need to ship any change anywhere?

**Comment by DXTimer** [comment]
We will need to update upstream e.g. `erldns`. Otherwise, no additional changes are required.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -40,7 +40,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        otp_version: ['26.1', '25.3', '24.3']
+        otp_version: ['26.2', '25.3']
```
Comment (weppos) [inline .github/workflows/ci.yml:43]: @whatyouhide do you want to test on 27 too? Or it's too big of a jump for this PR.

Comment (whatyouhide) [reply .github/workflows/ci.yml:43]: I wanted to do that in a separate PR to avoid mixing things up.

**Comment by whatyouhide** [comment]
Since this is a library (and not a deployable artifact in itself) I'll merge the PR.

**Pull request merged**